src/Integration/Binding/BindingProcessImpl.cs

```diff
@@ -97,52 +97,78 @@ internal class BindingProcessImpl : IBindingProcess
-                    return false;
+                    continue; // skip to the next language
                 }
-                this.InternalState.QualityProfiles[language] = qualityProfileInfo;
+
+                InternalState.QualityProfiles[language] = qualityProfileInfo;
 
                 var bindingConfiguration = QueueWriteBindingInformation();
 
                 // Create the binding configuration for the language
-                var bindingConfig = await this.bindingConfigProvider.GetConfigurationAsync(qualityProfileInfo, language, bindingConfiguration, cancellationToken);
+                var bindingConfig = await bindingConfigProvider.GetConfigurationAsync(qualityProfileInfo, language, bindingConfiguration, cancellationToken);
                 if (bindingConfig == null)
                 {
-                    this.host.Logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
+                    host.Logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
                         string.Format(Strings.FailedToCreateBindingConfigForLanguage, language.Name)));
                     return false;
                 }
 
-                this.InternalState.BindingConfigs[language] = bindingConfig;
+                InternalState.BindingConfigs[language] = bindingConfig;
 
-                currentLanguage++;
-                progress?.Report(new FixedStepsProgress(string.Empty, currentLanguage, languageCount));
-
-                this.host.Logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
+                host.Logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
                     string.Format(Strings.QualityProfileDownloadSuccessfulMessageFormat, qualityProfileInfo.Name, qualityProfileInfo.Key, language.Name)));
             }
 
             return true;
         }
 
+        /// <summary>
+        /// Attempts to fetch the QP for the specified language.
+        /// </summary>
+        /// <returns>The QP, or null if the language plugin is not available on the server</returns>
+        private async Task<SonarQubeQualityProfile> TryDownloadQualityProfileAsync(Language language, CancellationToken cancellationToken)
+        {
+            // There are valid scenarios in which a language plugin will not be available on the server:
+            // 1) the CFamily plugin does not ship in Community edition (nor do any other commerical plugins)
+            // 2) a recently added language will not be available in older-but-still-supported SQ versions
+            //      e.g. the "secrets" language
+            // The unavailability of a language should not prevent binding from succeeding.
+
+            // Note: the historical check that plugins meet a minimum version was removed. 
+            // See https://github.com/SonarSource/sonarlint-visualstudio/issues/4272
+
+            var qualityProfileInfo = await WebServiceHelper.SafeServiceCallAsync(() =>
```

Comment:
This logic hasn't changed; it was just moved to a helper method.
The comment above is new.